src/library_wasmfs.js

```diff
@@ -86,8 +86,8 @@ mergeInto(LibraryManager.library, {
 
       // Copy the file into a JS buffer on the heap.
       var buf = __wasmfs_read_file(pathName);
-      // The integer length is returned in the first 8 bytes of the buffer.
-      var length = {{{ makeGetValue('buf', '0', 'i64') }}};
+      // The integer length resides in the first 8 bytes of the buffer.
+      var length = {{{ makeGetValue('buf', '0', 'i32') }}};
```

Comment:
Hmm, how about changing

https://github.com/emscripten-core/emscripten/blob/ae675c6abdd7e4a73dfc100b7cd258ef0cec01a2/system/lib/wasmfs/js_api.cpp#L37

to write an i64? I think that might be better. The size can be more than 32 bits, and we are all set up for that, just the wrong type is written in that C++.

Reply:
Yeah, WasmFS currently uses 32-bit sizes in a lot of places where it should be using 64-bit sizes...

Reply:
Commit 567cfd3cf97835be64c482866ac21523ad6ca68d ensures that it writes an i64 in C++, and that it is read out as an i64 in JS. I've updated the PR description accordingly.